code/obj/flock/cage.dm

```diff
@@ -122,72 +186,118 @@
+
+		UpdateIcon()
+
 		if(occupant)
-			underlays -= occupant
-			underlays += occupant
 			if(eating_occupant && prob(20))
 				boutput(occupant, "<span class='flocksay italics'>[pick_string("flockmind.txt", "flockmind_conversion")]</span>")
-		if(src.contents.len <= 0 && reagents.get_reagent_amount(target_fluid) < 50)
-			if(reagents.has_reagent(target_fluid)) // flood the area with our unprocessed contents
+		if(src.contents.len <= 0 && reagents.get_reagent_amount(target_fluid) < create_egg_at_fluid)
+			if(reagents.has_reagent(target_fluid)) // dump out our excess resources as a cache
 				playsound(src, "sound/impact_sounds/Slimy_Splat_1.ogg", 80, 1)
-				T.fluid_react_single(reagents.get_reagent_amount(target_fluid))
+				var/obj/item/flockcache/x = new(src.loc)
+				x.resources = reagents.get_reagent_amount(target_fluid)
+				reagents.del_reagent(target_fluid,x.resources)
 			qdel(src)
 
 	disposing()
 		playsound(src, "sound/impact_sounds/Energy_Hit_2.ogg", 80, 1)
-		processing_items -= src
-		occupant?.removeOverlayComposition(/datum/overlayComposition/flockmindcircuit)
+
+		if(istype(occupant,/mob/living))
+			var/mob/living/M = occupant
+			M?.removeOverlayComposition(/datum/overlayComposition/flockmindcircuit)
+
+		for(var/atom/movable/AM in src)
+			if(ismob(AM))
+				var/mob/M = AM
+				M.visible_message("<span class='alert'><b>[M]</b> breaks out of [src]!</span>","<span class='alert'>You break out of [src]!</span>")
+			AM.set_loc(src.loc)
 		..()
 
-/obj/icecube/flockdrone/special_desc(dist, mob/user)
-	if(isflock(user))
-		return {"<span class='flocksay'><span class='bold'>###=-</span> Ident confirmed, data packet received.
-		<br><span class='bold'>ID:</span> Matter Reprocessor
-		<br><span class='bold'>System Integrity:</span> [round((src.health/src.health_max)*100)]%
-		<br><span class='bold'>Volume:</span> [src.reagents.get_reagent_amount(src.target_fluid)]
-		<br><span class='bold'>Needed volume:</span> [src.create_egg_at_fluid]
-		<br><span class='bold'>###=-</span></span>"}
-	else
-		return null // give the standard description
-
-/obj/icecube/flockdrone/attack_hand(mob/user as mob)
-	if (user.a_intent == INTENT_HARM)
-		user.visible_message("<span class='alert'><b>[user]</b> kicks [src]!</span>", "<span class='alert'>You kick [src]!</span>")
-		attack_particle(user, src)
-		user.lastattacked = src
-		takeDamage(2)
-		playsound(src, "sound/impact_sounds/Crystal_Hit_1.ogg", 25, 1)
-
-/obj/icecube/flockdrone/attackby(obj/item/W as obj, mob/user as mob)
-	user.visible_message("<span class='alert'><b>[user]</b> hits [src] with [W]!</span>", "<span class='alert'>You hit [src] with [W]!</span>")
-	attack_particle(user, src)
-	user.lastattacked = src
-	takeDamage(W.force)
-	playsound(src, "sound/impact_sounds/Crystal_Hit_1.ogg", 25, 1)
+	relaymove(mob/user as mob)
+		if (user.stat)
+			return
+
+		if(prob(25))
+			takeDamage("brute",1)
+		return
+
+	takeDamage(var/damageType, var/amount)
+		..(damageType,amount)
```

Comment:
Since this is calling the takeDamage() of /obj/flock_structure, checkHealth() should be overridden I think to just have a body of return, otherwise it's going to create glass shards and stuff on destruction, when in reality the Flock cage has its own way of destruction.